docs/verification.md

```diff
@@ -0,0 +1,71 @@
+# Verifying Trusted Setup Ceremony
+
+A trusted setup ceremony is a fundamental process in ensuring the integrity and security of certain cryptographic systems. 
+By generating a unique and unpredictable “number” collectively, with contributions from multiple parties, 
+we create a foundation of trust that prevents malicious actors from compromising the system.
+
+Verifying the security of our ceremony is paramount. However, it's important to note that it's a more technical process than contributing. 
+We invite everyone to confirm the security of our ceremony for themselves, and so are publishing this guide to help you do so.
+
+
+## Overview
+There are multiple steps to verifying the security of a trusted setup ceremony. We provide the required list in a way 
+that malicious projects pretend to run rituals while secretly retaining backdoors. 
+To ensure that anyone can verify the integrity of our trusted setup ceremony, we have outlined a robust guide for 
+verifying the results. Here’s an overview of the key steps involved:
+
+- Verifying the circuit we are securing does not include security holes;
+- Verifying the transcript matches the circuit;
+- Verifying contributors' attestations match the transcript;
+
+## Verifying the Circuit
+Confirming the integrity and security of the [EthStorage blob poseidon circuit](https://github.com/ethstorage/zk-decoder/blob/main/circom/circuits/blob_poseidon_2.circom) 
+by reviewing our circom code, which is open-source and available on Github.
+
+## Verifying the Transcript Matches the Circuit
+Our ceremony transcript is included in the zkey published on ceremony.ethstorage.io in the "Download ZKey" tab of the 
+[EthStorage Circom page](https://ceremony.ethstorage.io/projects/EthStorage%20circom). You can verify it matches the 
+circuit using Circom and snarkjs:
+
+- Install [Circom](https://docs.circom.io/getting-started/installation/) and [snarkjs](https://github.com/iden3/snarkjs).
+- Clone circom code https://github.com/ethstorage/zk-decoder.
+- Download the Powers of Tau (ptau) file we used for our ceremony; we used the Hermez rollup with 2^20 powers of tau, 
+which is linked in [the snarkjs readme](https://github.com/privacy-scaling-explorations/perpetualpowersoftau?tab=readme-ov-file#prepared-phase-2-files) 
+or available directly [here](https://pse-trusted-setup-ppot.s3.eu-central-1.amazonaws.com/pot28_0080/ppot_0080_20.ptau).
+- Generate the `r1cs` file from the `circom` source files, by running `circom blob_poseidon_2.circom --r1cs` in the 
+directory `zk-decoder/circom/circuits` (need to run `npm install` for the first time). The SHA-256 hash of this file
```

Comment:
It's confusing about where to run `npm install`

Reply:
under zk-decoder/circom/circuits, same folder of `circom blob_poseidon_2.circom --r1cs`

Reply:
Actually `npm install` should be run under `zk-decoder/circom` and it should be told clearly

Reply:
both `zk-decoder/circom` and `zk-decoder/circom/circuits` are ok

Reply:
ok